src/utils/server/withServerActionMiddleware.ts

```
@@ -30,7 +34,20 @@ export function withServerActionMiddleware<T extends (...args: any) => any>(
         headers: headers(),
         formData: convertArgsToFormData(args),
       },
-      () => action(...args),
+      () => {
+        const promise = action(...args)
+        promise
+          .then((result: any) => {
+            if (result === undefined) {
+              throw new Error('Undefined result')
+            }
+          })
+          .catch(function (e: any) {
+            return logger.error(e)
+          })
+
+        return promise
```

Few things to note here:
1. Putting a `.catch` suppresses all internal server errors the actions might want to throw, you are logging and then returning `undefined` for the request 
2. Since you're suppressing errors (including the one you are throwing in line 42) this request will continue to respond with undefined. If the idea is to fail the request if the result is `undefined` removing the `.catch` will work properly
3. If you wanna log the error anyways, just throw the error inside the `.catch` scope
```typescriptreact
.catch(error => {
  logger.error(error)
  throw error
})
```
4. On the `.catch` syntax, we shouldn't use anonymous functions with the `function` keyword (they should be arrow functions) and you don't need to specify a type for the error.
5. You can do all of this inline, `.then` and `.catch` return the promise itself, you can do `return action(...).then(...).catch(...)` normally

---

yes, this is why I found my solution weird, but without the catch, I receive an  error that I am throwing an error on an async function without a catch and got unhandled error in Sentry. So in the end I remove one error by replacing by another one... 😕
For the 4th point, I end up using an anonymous function because my IDE keep complaining about that I didn't specify an error type 🤷‍♂️

Anyway I am going to try couple of your suggestion